### PR TITLE
Update to latest ledgerjs and pull in speculos transport

### PR DESCRIPTION
**Pull request by 3noch**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "private": true,
   "scripts": {
-    "preinstall": "git submodule update --recursive --init && cd ledgerjs && yarn install && yarn build",
+    "preinstall": "[ \"$(ls -A ledgerjs)\" ] || git submodule update --recursive --init ledgerjs; cd ledgerjs; yarn install --frozen-lockfile && yarn build",
```
Comment (3noch) [inline package.json:4]: This ensures we don't overwrite changes inside the submodule.

Comment (lpsmith) [reply package.json:4]: Good, I'd deleted the `git submodule update` but was keeping it out of commits

**Comment by 3noch** [comment]
@lpsmith I just added speculos support.

**Comment by lpsmith** [comment]
Why does the `packages/hw-app-ckb/src/Ckb.js` file even exist?

I mean, I'm not too worried... My biggest concern would be that maybe it'll get included in the final bundle... which wouldn't be ideal.

**Comment by 3noch** [comment]
Probably because yarn found it in the ledgerjs repo and we're based on nervos branch.

File: cli/cli.js [hunk 2]
```diff
@@ -9,7 +9,9 @@ const BN = require("bn.js");
 const URI = require("urijs");
 const commander = require("commander");
 const AvaJS = require("avalanche");
+const { command } = require("commander");
```
Comment (3noch) [inline cli/cli.js:12]: This seems like a mistake

**Comment by 3noch** [comment]
@lpsmith Where are you seeing that? Is it in the yarn.lock file?

File: cli/cli.js [hunk 3]
```diff
@@ -40,11 +42,24 @@ commander.Command.prototype.add_node_option = function() {
   return this.option("-n, --node <uri>", "node to use", "https://testapi.avax.network");
 }
 
+commander.Command.prototype.add_speculos_option = function() {
+  return this.option("--speculos <apdu-port>", "(for testing) use the Ledger Speculos transport instead of connecting via USB; overrides --device", parseInt);
```
Comment (tomsmalley) [inline cli/cli.js:45]: does this belong in add_device_option, or are there cases where we use the ledger device without wanting speculos support?

Comment (3noch) [reply cli/cli.js:45]: There is exactly one case where --speculos is basically useless: get-device-model. That just prints `null` for speculos. But that's probably not a good enough reason to keep these apart now that I think about it.

Comment (3noch) [reply cli/cli.js]: Done

**Comment by lpsmith** [comment]
Oh, that's not a file in this repo itself,  it's just github's diff tool getting fancy.

**Pull request merged**